Subsurface/FlightComputer/include/NetClient.h

```diff
@@ -0,0 +1,236 @@
+#pragma once
+#define ASIO_STANDALONE
+#include <asio.hpp>
+
+#include "Message.h"
+#include "Paradigm.pb.h"
+
+#include <deque>
+#include <iostream>
+
+using asio::ip::tcp;
+using asio::ip::udp;
+
+template<typename Proto>
+class TcpClient
+{
+public:
+	TcpClient(asio::io_context& ioContext, const tcp::resolver::results_type& endpoints)
+		: m_ioContext(ioContext), m_socket(ioContext)
+	{
+		connect(endpoints);
+	}
+
+	void write(const Message<Proto>& message)
+	{
+		asio::post(m_ioContext,
+			[this, message]()
+			{
+				bool writeInProgress = !m_writeMessages.empty();
+				m_writeMessages.push_back(message);
+				if (!writeInProgress)
+				{
+					internalWrite();
+				}
+			});
+	}
+
+	bool connected() const { return m_isConnected; }
+
+	void close()
+	{
+		asio::post(m_ioContext, [this]() {m_socket.close(); });
+		m_isConnected = false;
+	}
+
+	const Message<Proto>& getReadMessage() { return m_readMessage; }
+
+private:
+	void connect(const tcp::resolver::results_type& endpoints)
+	{
+		asio::async_connect(m_socket, endpoints,
+			[this, &endpoints](std::error_code ec, tcp::endpoint)
+			{
+				if (!ec)
```

Comment:
Compare against error code value

Reply:
I think the reason this is done this way is because error codes are platform-dependent, and not sure that there is an error code indicating success.